supervisor/docker/progress.py

```diff
@@ -0,0 +1,132 @@
+"""Docker pull progress emitter."""
+import asyncio
+import time
+from contextlib import suppress
+
+import docker
+
+from ..coresys import CoreSysAttributes
+from ..exceptions import HomeAssistantAPIError
+
+
+def pull_with_progress(interface: CoreSysAttributes, image, tag):
```

Comment:
Implement this as part of the DockerAPI object and don't forward the interface, lookup it over the coresys

Reply:
Do you mean the whole file or just this function?

Reply:
this function, move the states and the other object into docker/utils.py as helper like we do in other modules

Reply:
OK, will try it maybe tomorrow

Reply:
@pvizeli pls check